File: src/main/java/net/gtaun/shoebill/common/command/Command.java
package net.gtaun.shoebill.common.command;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD})
public @interface Command
{
	String name() default "";
	short priority() default 0;
	boolean caseSensitive() default false;
}

File: src/main/java/net/gtaun/shoebill/common/command/Command.kt
package net.gtaun.shoebill.common.command


@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class Command(val name: String = "", val priority: Short = 0, val caseSensitive: Boolean = false)
